elasticdl/python/master/task_dispatcher.py

```diff
@@ -207,7 +219,10 @@ def report(self, task_id, success):
                 logger.warning("Unknown task_id: %d" % task_id)
             elif not success:
                 # TODO: keep count of retries.
-                self._todo.append(task)
+                if task.type == elasticdl_pb2.TRAINING:
```

Comment:
I think we need check task.type == elasticdl_pb2.EVALUATION, otherwise we will add the task into _todo.
Because we treat TRAINING/PREDICTION/SAVE_MODEL task in the same way.